android/src/main/java/com/rnldk/RnLdkModule.kt

```diff
@@ -191,23 +202,93 @@ class RnLdkModule(private val reactContext: ReactApplicationContext) : ReactCont
       channelMonitors = channel_monitor_list.toTypedArray();
     }
 
+    // initialize graph sync #########################################################################
+
+    if (networkGraphPath != "") {
+      println("ReactNativeLDK: using network graph path: $networkGraphPath");
+      val f = File(networkGraphPath);
+      if (f.exists()) {
+        println("ReactNativeLDK: loading network graph...");
+        val serialized_graph = File(networkGraphPath).readBytes()
+        val readResult = NetworkGraph.read(serialized_graph)
+        if (readResult is Result_NetworkGraphDecodeErrorZ.Result_NetworkGraphDecodeErrorZ_OK) {
+          router = readResult.res
+          println("ReactNativeLDK: loaded network graph ok")
+        } else {
+          println("ReactNativeLDK: network graph load failed")
+          if (readResult is Result_NetworkGraphDecodeErrorZ.Result_NetworkGraphDecodeErrorZ_Err) {
+            println("ReactNativeLDK: " + readResult.err);
+          }
+          // error, creating from scratch
+          println("ReactNativeLDK: network graph error, creating from scratch")
+          router = NetworkGraph.of(hexStringToByteArray("000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f").reversedArray())
+        }
+      } else {
+        // first run, creating from scratch
+        println("ReactNativeLDK: network graph first run, creating from scratch")
+        router = NetworkGraph.of(hexStringToByteArray("000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f").reversedArray())
+      }
+
+      scorer = MultiThreadedLockableScore.of(Scorer.with_default().as_Score())
```

Comment:
You probably want to persist the scorer as well.

Reply:
how much space will it take?

Reply:
10s of bytes per node on the network, max, so smaller than the graph, but it only grows when it sees failures/successes on different nodes.

Reply:
why not keep scores as part of the graph?

Reply:
Some users may sync the data from different places (p2p vs server, cause servers have more info, etc), plus we want to support multiple different scorers, or user custom scorers.